File: 프로그래머스/해시/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> m;
    
    for(string name : participant)
        ++m[name];
    
    for(string name : completion)
        --m[name];
    
    for(auto pair : m){
        if(pair.second > 0)
            return pair.first;
    }
}

File: 프로그래머스/해시/완주하지 못한 선수.py
from collections import defaultdict

def solution(participant, completion):
    answer = ''
    mydict = defaultdict(int)
    
    for i in completion:
        mydict[i] += 1
    
    for i in participant:
        if not mydict[i]:
            answer = i
            break
        mydict[i] -= 1
        
    return answer
